### PR TITLE
Chore: Avoid explicit `React.FC<Props>` when possible

### PR DESCRIPTION
**Pull request by ryantxu**
This PR replaces most explicit uses of React.FC with a typed function.

The explicit FC components exist everywhere so are often used as the copy from source :(

See:
* https://medium.com/raccoons-group/why-you-probably-shouldnt-use-react-fc-to-type-your-react-components-37ca1243dd13
* https://github.com/facebook/create-react-app/pull/8177

### CODE REVIEW

File: packages/grafana-toolkit/src/cli/templates/component.tsx.template.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
-export const componentTpl = `import React, { FC } from 'react';
+export const componentTpl = `import React from 'react';
 
 export interface Props {};
 
-export const <%= name %>: FC<Props> = (props) => {
+export const <%= name %> = (props: Props) => {
```
Comment (ryantxu) [inline packages/grafana-toolkit/src/cli/templates/component.tsx.template.ts:5]: @mckn -- should I pull out the toolkit changes?  or does it seem OK as part of this PR?

Comment (jackw) [reply packages/grafana-toolkit/src/cli/templates/component.tsx.template.ts:5]: I think it's fine to keep it. 👍 
FYI - we've [deprecated the command that uses them](https://github.com/grafana/grafana/blob/main/packages/grafana-toolkit/src/cli/index.ts#L87-L102) with the intention to remove it (and these templates) completely in Grafana 10 in favour of [react-gen-component](https://www.npmjs.com/package/react-gen-component).

**Review by jackw — APPROVED** [review]
Lovely stuff! 🚀

**Review by mckn — APPROVED** [review]
Great work on this!

**Review by nmarrs — APPROVED** [review]
✨

**Pull request merged**